File: bojaxz-cs50-problems-2020-x-sentimental-mario-more/mario.py
# import the get_int function from cs50 library
from cs50 import get_int

# define get height funtion that will get a height from the user betweem 1 and 8 inclusive.
# then reprompt the user if number is out of bounds


def get_height():
    while True:
        try:
            n = get_int("Please enter height:\n")
            if 0 < n < 9:
                return n
            else:
                print("Number not in range.")
        except ValueError:
            print("Invalid Input")

# function to build the pyramid using the int from the user in get_height


def height():
    pheight = get_height()
    h = 0
    while h < pheight:
        h += 1
        print((" " * (pheight - h)), end="")
        print(("#" * (h)), end="  ")
        print("#" * (h))


height()

File: bojaxz-cs50-problems-2020-x-mario-less/mario.c
#include <cs50.h>
#include <stdio.h>
//mario based programme that will create the steps at the end of a mario level based on the users input.
int main(void)
{
    //prompt user for height
    int height;
    do
    {
        height = get_int("Height: \n");
    }
    while (height < 1 || height > 8);
    
    //build stairs according to height
    for (int x = 0; x < height; x++)
    {
        for (int z = 0; z < height - x - 1; z++)
        {    
            printf(" ");
        }
        for (int y = 0; y <= x; y++)
        {
            printf("#");
        }
        printf("\n");
    }
}